resources/assets/components/CampaignInbox/index.js

```diff
@@ -70,6 +71,29 @@ class CampaignInbox extends React.Component {
     });
   }
 
+  updateTag(postId, tag) {
+    const fields = {
+      post_id: postId,
+      tag_name: tag,
+    };
+
+    let response = this.api.post('tags', fields);
+    response.then((data) => {
+      console.log(data);
+      this.setState((previousState) => {
+        const newState = {...previousState};
+        const user = newState.posts[postId].user;
+
+        newState.posts[postId] = data;
+
+        // Keep the user from the initial page load.
+        newState.posts[postId].user = user;
```

Comment:
the user doesn't remain in the state: 
![screen shot 2017-05-24 at 3 21 57 pm](https://cloud.githubusercontent.com/assets/9019452/26421723/0628e3fc-4095-11e7-80e2-7b8a85da0d67.png)